### PR TITLE
Initial PR 

### PR DESCRIPTION
**Pull request by LaurentLesle**
Add kusto cluster and database

### CODE REVIEW

File: examples/cluster_with_databases/main.tf [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = ">= 3.5.0, < 4.0.0"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
```
Comment (segraef) [inline examples/cluster_with_databases/main.tf]: Set `prevent_deletion_if_contains_resources` to `false` in provider block in example code per spec [TFNFR36](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr36---category-code-style---example-code-must-set-prevent_deletion_if_contains_resources-to-false-in-provider-block)

File: examples/default/terraform.tfvars
Comment (segraef) [inline examples/default/terraform.tfvars:1]: remove empty files

File: examples/default/.gitkeep
Comment (segraef) [inline examples/default/.gitkeep:1]: empty files not required

File: examples/default/main.tf
Comment (segraef) [inline examples/default/main.tf:1]: `provider` block missing which also means to set `prevent_deletion_if_contains_resources` to `false` in provider block in example code per spec [TFNFR36](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr36---category-code-style---example-code-must-set-prevent_deletion_if_contains_resources-to-false-in-provider-block)

File: examples/.gitignore
Comment (segraef) [inline examples/.gitignore:1]: move to global `.gitignore` if required

File: terraform.tf [hunk 2]
```diff
@@ -1,10 +1,9 @@
 terraform {
-  required_version = ">= 1.3.0"
+  required_version = ">= 1.6.0"
   required_providers {
-    # TODO: Ensure all required providers are listed here.
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = ">= 3.71.0"
+      version = ">= 3.88.0"
```
Comment (segraef) [inline terraform.tf:6]: Consider setting a constraint on maximum major version of Provider as per spec [TFNFR26](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr26---category-code-style---provider-version-constraint-must-have-a-constraint-on-maximum-major-version).

File: examples/default/main.tf [hunk 3]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.3.0"
+  required_version = ">= 1.7.0"
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
```
Comment (segraef) [inline examples/default/main.tf:5]: Consider setting a constraint on maximum major version of Provider as per spec [TFNFR26](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr26---category-code-style---provider-version-constraint-must-have-a-constraint-on-maximum-major-version).

File: modules/azurerm_kusto_database_principal_assignment/outouts.tf
Comment (segraef) [inline modules/azurerm_kusto_database_principal_assignment/outouts.tf:1]: typo in file

**Review by segraef — DISMISSED** [review]
Thanks for the great work @LaurentLesle ! It looks OK so far apart from my annotations. Let me me trigger @mbilalamjad who will start the official AVM-review so we can get your awesome module published 🚀

File: examples/cluster_with_databases/main.tf (hunk 1 above)
Comment (LaurentLesle) [reply examples/cluster_with_databases/main.tf]: Don't remembered the linter complaining against it...

File: examples/.gitignore
Comment (LaurentLesle) [reply examples/.gitignore:1]: Nope I only need to negate the exclusion of the *.tfvars. I am using them for the terraform tests in the tests folder by using the tfvars in the example folder.
Initial implementation of terraform test.

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "image": "mcr.microsoft.com/azterraform:latest",
+  "image": "ubuntu",
```
Comment (segraef) [inline .devcontainer/devcontainer.json:2]: intentionally?

**Review by segraef — COMMENTED** [review]
Thank you for this massive module @LaurentLesle, for me everything looks fine but now over to @mbilalamjad or @matt-FFFFFF for the final review.

File: .devcontainer/devcontainer.json (hunk 4 above)
Comment (LaurentLesle) [reply .devcontainer/devcontainer.json:2]: Yeah I could not get az cli working to test the deployment in Azure. So I switched to ubuntu

**Comment by LaurentLesle** [comment]
@segraef @mbilalamjad @matt-FFFFFF any chance to review the PR?

**Pull request merged**